File: app/src/main/java/com/lostincontext/commons/list/SectionViewHolder.java
package com.lostincontext.commons.list;


import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.lostincontext.databinding.ItemSectionHeaderBinding;

public class SectionViewHolder extends ViewHolder {


    private ItemSectionHeaderBinding binding;

    public static SectionViewHolder create(LayoutInflater layoutInflater,
                                           ViewGroup parent) {
        ItemSectionHeaderBinding headerBinding = ItemSectionHeaderBinding.inflate(layoutInflater,
                                                                                  parent,
                                                                                  false);
        return new SectionViewHolder(headerBinding);
    }


    public SectionViewHolder(ItemSectionHeaderBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }


    public void bindTo(Section section) {
        binding.setTitle(section.getTitle());
    }

}

File: app/src/main/java/com/lostincontext/commons/list/SectionViewHolder.kt
package com.lostincontext.commons.list


import android.view.LayoutInflater
import android.view.ViewGroup

import com.lostincontext.databinding.ItemSectionHeaderBinding

class SectionViewHolder(private val binding: ItemSectionHeaderBinding) : ViewHolder(binding.root) {


    fun bindTo(section: Section<*>) {
        binding.title = section.title
    }

    companion object {

        fun create(layoutInflater: LayoutInflater,
                   parent: ViewGroup): SectionViewHolder {
            val headerBinding = ItemSectionHeaderBinding.inflate(layoutInflater,
                                                                 parent,
                                                                 false)
            return SectionViewHolder(headerBinding)
        }
    }

}
